addons/website/static/src/components/dialog/page_properties.js

```diff
@@ -155,108 +154,122 @@ DuplicatePageDialog.template = xml`
 </WebsiteDialog>
 `;
 DuplicatePageDialog.props = {
-    onClose: Function,
-    close: Function,
+    onDuplicate: {type: Function, optional: true},
     pageId: Number,
+    close: Function,
+};
+
+export const pagePropertiesProps = {
+    title: _t('Page Properties'),
+    resModel: 'website.page',
+    context: {
+        form_view_ref: 'website.website_page_properties_view_form',
+    },
 };
 
-export class PagePropertiesDialogWrapper extends Component {
+export class PagePropertiesDialog extends FormViewDialog {
     setup() {
-        try {
-            this.websiteService = useService('website');
-            this.dialogService = useService('dialog');
-            this.orm = useService('orm');
-        } catch {
-            // Use services in legacy environment.
-            this.websiteService = useWowlService('website');
-            this.dialogService = useWowlService('dialog');
-            this.orm = useWowlService('orm');
-        }
+        super.setup();
+        this.dialog = useService('dialog');
+        this.website = useService('website');
 
-        onRendered(this.createDialog);
-    }
+        this.viewProps.resId = this.resId;
+        this.viewProps.buttonTemplate = 'website.PagePropertiesDialog.Buttons';
 
-    createDialog() {
-        if (this.props.mode === 'clone') {
-            this.dialogService.add(DuplicatePageDialog, {
-                pageId: this.pageId,
-                onClose: this.props.onClose,
-            });
-        } else if (this.props.mode === 'delete') {
-            this.dialogService.add(DeletePageDialog, {
-                pageId: this.pageId,
-                onClose: this.props.onClose,
-            });
-        } else {
-            const parent = standaloneAdapter({Component});
-            const formViewDialog = new FormViewDialog(parent, this.dialogOptions);
-            formViewDialog.buttons = [...formViewDialog.buttons, ...this.extraButtons];
-            formViewDialog.on('form_dialog_discarded', parent, () => this.websiteService.context.showPageProperties = false);
-            formViewDialog.open();
-        }
+        onMounted(() => {
+            this.actionClone = this.modalRef.el.querySelector('.o_page_properties_clone');
+            this.actionDelete = this.modalRef.el.querySelector('.o_page_properties_delete');
+            this.actionClone.addEventListener('click', this.clonePage.bind(this));
+            this.actionDelete.addEventListener('click', this.deletePage.bind(this));
+        });
+
+        onWillUnmount(() => {
+            this.actionClone.removeEventListener('click', this.clonePage.bind(this));
+            this.actionDelete.removeEventListener('click', this.deletePage.bind(this));
+        });
```

Comment:
Should not this be done via a `t-on-click` in the OWL template? And would mean 0-2 XML lines instead of 11 JS lines? (real question, I am a OWL noob)

Reply:
Actually, this is a special case:
The `FormViewDialog` component (which is extended by `PagePropertiesDialog`) has the `View` as sub-component, so the `buttonTemplate` prop (footer actions template) is passed to the view to override the original one used by formViews (`t-name="web.FormView.Buttons"` [A]), so to have the new page properties buttons (Duplicate/Delete page), I had to pass the new XML in `buttonTemplate`, which means I can't use `t-on-click` simply because I can't handle them outside of `PagePropertiesDialog`, ([A] is linked to the formView), so I had to handle them directly from the DOM, I discussed this With ARD last time and he suggested to use the `useListener` hook (which does the same thing as the current code in a hook) but after checking the code, it seems [deprecated](https://github.com/odoo/odoo/blob/8a5721bc7d5540e382d2d06fe5b9634911b7859c/addons/web/static/src/core/utils/hooks.js#L86) since it uses the old `component.el` attribute (details [here](https://github.com/odoo/odoo/commit/8a5721bc7d5540e382d2d06fe5b9634911b7859c)).

Reply:
I see. I guess you can't use t`-on-click` directly but what you do here seems a bit breaking the OWL logic IMO: you have a component (a "dialog") which contains another (a "view") but the "dialog", once mounted, search the "view" DOM for stuff and add event listeners. Should not you rather give props (from the dialog to the view) which are the handlers to call for the buttons? So in your button template, something like `t-on-click="props.clonePage"` (via `viewProps` that I see ?). If not that, a variant of it. I am sure there is a way for the button section to "communicate" with the dialog component (either via props that are given by the dialog as I explained or another system) but I doubt the framework team would approve of searching the DOM of a subcomponent... :grimacing: In doubt, check with them, they don't bite (I think) :wink: (cc @detrouxdev as you were mentioned)

Reply:
I updated the code to override the `FormViewDialog` related template and add buttons directly in the footer's slot (I just forgot that I have this possibility) Thanks :+1: